sensing/tier4_pcl_extensions/include/tier4_pcl_extensions/voxel_grid_nearest_centroid_impl.hpp

```diff
@@ -320,19 +322,15 @@ void pcl::VoxelGridNearestCentroid<PointT>::applyFilter(PointCloud & output)
 
       output.push_back(PointT());
 
-      // TODO(kenji.miyake) refactor
-      double dis_min = 1000000.0;
-      size_t i_min = 0;
-      for (size_t i = 0; i < leaf.points.size(); ++i) {
-        double dis = std::pow(leaf.points.at(i).x - leaf.centroid[0], 2.0) +
-                     std::pow(leaf.points.at(i).y - leaf.centroid[1], 2.0) +
-                     std::pow(leaf.points.at(i).z - leaf.centroid[2], 2.0);
-        if (dis < dis_min) {
-          dis_min = dis;
-          i_min = i;
-        }
-      }
-      output.points.back() = leaf.points.at(i_min);
+      const auto & centroid = leaf.centroid;
+      const auto distances = leaf.points | ranges::views::transform([centroid](const auto p) {
```

Comment:
Using references is better? :thinking: 

```suggestion
      const auto distances = leaf.points | ranges::views::transform([&centroid](const auto & p) {
```

Reply:
fixed 0a6c888 .